src/lib.rs

```diff
@@ -270,4 +280,63 @@ where
         let result: i16 = (bytes[1] as i16) << 8 | bytes[0] as i16;
         Ok((result as f32) / TEMP_SCALE + TEMP_BIAS)
     }
+
+    // --- ADDED GROUP OF FUNCTIONS ---
+    
+    /// Read a byte from the given register.
+    // fn read_register(&mut self, sensor: Sensor, address: u8) -> Result<u8, T::Error> {
+    pub fn read_register(&mut self, sensor: Sensor, address: u8) -> Result<u8, T::Error> {
+        let mut reg_data = [0u8];        
+        self.interface.read(sensor, address, &mut reg_data)?;
+        Ok(reg_data[0])
+    }
+
+    
+    /// Check if specific bits are set.
+    fn is_register_bit_flag_high(&mut self, sensor: Sensor, address: u8, bitmask: u8,) -> Result<bool, T::Error> {
+        let data = self.read_register(sensor, address)?;
+        Ok((data & bitmask) != 0)
+    }
+    
 }
+
+/// Settings for various FIFO- and interrupt-related flags, Enabled or Disabled
+#[allow(non_camel_case_types)]
+#[derive(Debug, Clone, Copy)]
+pub enum FLAG {
+    /// Enabled (bit set)
+    Enabled,
+    /// Disabled (bit cleared)
+    Disabled,
+}
+
+impl FLAG {
+    pub fn status(self) -> bool {
+        let status = match self {
+            FLAG::Disabled => false,
+            FLAG::Enabled => true,
+        };
+        status
+    }
+}
```

Comment:
OK, I see what you mean. I wanted it to be very explicit that it can only be on or off, but of course having an enum nicely solves that anyway. So how does it work now? I would need to change the FLAG enum, and also fix every function that uses it, right?

Reply:
I see what you mean now. OK, this will take some time to fix, as FLAG is used accross all the interrupt modules. 
If that is the way to do it, I suppose it should also be done for other similar enums (`INT_ACTIVE`, I`NT_PIN`, `INT_LATCH`, `POS_RECOG`, `COUNTER` in `interrupts/mod.rs`, and also `COMBINATION` in `lib.rs`)?